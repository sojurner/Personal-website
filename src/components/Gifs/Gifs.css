.gif-items {
}

#scrolling {
  height: 400px;
  width: 400px;
  overflow-y: auto;
}
