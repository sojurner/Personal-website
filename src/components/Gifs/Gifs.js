import React, { Component } from 'react';
import * as calls from '../../utilities/apiCalls';
import InfiniteScroll from 'react-infinite-scroll-component';

import './Gifs.css';

class Gifs extends Component {
  constructor() {
    super();
    this.state = {
      gifs: null,
      input: '',
      query: '',
      mappedGifs: [],
      loading: true,
      start: 0,
      end: 15,
      searchResults: ''
    };
  }

  componentDidMount() {
    this.getGifs();
  }

  getGifs = async () => {
    const { start, end } = this.state;
    const gifs = await calls.getGifs('paul');
    this.setState({
      gifs: gifs.data,
      mappedGifs: gifs.data.slice(start, end),
      loading: false
    });
    // this.renderGifs();
  };

  handleChange = event => {
    this.setState({ input: event.target.value });
  };

  searchGifs = async e => {
    e.preventDefault();
    const { query, start, end, input } = this.state;
    this.setState({ start: 0, end: 6 });
    const gifs = await calls.getGifs(input);
    this.setState({
      gifs: gifs.data,
      mappedGifs: gifs.data.slice(start, end),
      loading: false,
      query: input,
      input: ''
    });
  };

  renderGifs = async () => {
    let { start, end, gifs, mappedGifs } = this.state;
    await this.setState({
      start: end + 1,
      end: end + 14
    });
    await setTimeout(async () => {
      await this.setState({
        mappedGifs: [...mappedGifs, ...gifs.slice(start, end)]
      });
    }, 2000);
  };

  render() {
    const { query, mappedGifs, searchResults, input } = this.state;
    return (
      <div className="gif-container" id="gifs">
        <h1 className="jify">{query ? query.toUpperCase() : 'PAUL'} LAND</h1>
        <form className="gif-form" onSubmit={this.searchGifs}>
          <input
            className="gif-input"
            type="text"
            value={input}
            onChange={this.handleChange}
            placeholder="Search Gifs"
          />
          <i className="fab fa-searchengin" onClick={this.searchGifs} />
        </form>
        {query && (
          <h4 className="search-results">
            Search results for: '<strong>{query}</strong>'
          </h4>
        )}
        <div id="scrolling">
          <InfiniteScroll
            className="infinite-scroll"
            dataLength={mappedGifs.length}
            next={this.renderGifs}
            hasMore={true}
            loader={<h4>Loading...</h4>}
            scrollableTarget="scrolling"
          >
            {this.state.mappedGifs.map((gif, index) => {
              const url = `https://media.giphy.com/media/${gif.id}/giphy.gif`;
              return (
                <img
                  key={`gif-${index}`}
                  className="gif-items"
                  src={url}
                  height="200"
                  width="200"
                />
              );
            })}
          </InfiniteScroll>
        </div>
      </div>
    );
  }
}

export default Gifs;
