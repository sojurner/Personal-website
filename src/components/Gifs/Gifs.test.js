import React from 'react';
import Gifs from './Gifs';

describe('Gifs', () => {
  let wrapper;
  let mockGifs = [
    {
      type: 'gif',
      id: 'ZXdPWPdzDTFlK',
      slug: 'john-paul-george-ZXdPWPdzDTFlK',
      url: 'https://giphy.com/gifs/john-paul-george-ZXdPWPdzDTFlK',
      bitly_gif_url: 'https://gph.is/Ko8iP9',
      bitly_url: 'https://gph.is/Ko8iP9',
      embed_url: 'https://giphy.com/embed/ZXdPWPdzDTFlK',
      username: '',
      source: 'https://dreaming-in-the-60s.tumblr.com/post/43739719119',
      rating: 'g',
      content_url: '',
      source_tld: 'dreaming-in-the-60s.tumblr.com',
      source_post_url:
        'https://dreaming-in-the-60s.tumblr.com/post/43739719119',
      is_sticker: 0,
      import_datetime: '2014-01-03 23:41:17',
      trending_datetime: '2014-12-11 18:48:23',
      images: {
        fixed_height_still: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200_s.gif',
          width: '241',
          height: '200'
        },
        original_still: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy_s.gif',
          width: '275',
          height: '228'
        },
        fixed_width: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200w.gif',
          width: '200',
          height: '166',
          size: '786205',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200w.mp4',
          mp4_size: '64712',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200w.webp',
          webp_size: '198578'
        },
        fixed_height_small_still: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100_s.gif',
          width: '121',
          height: '100'
        },
        fixed_height_downsampled: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200_d.gif',
          width: '241',
          height: '200',
          size: '235844',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200_d.webp',
          webp_size: '54672'
        },
        preview: {
          width: '168',
          height: '138',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy-preview.mp4',
          mp4_size: '41804'
        },
        fixed_height_small: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100.gif',
          width: '121',
          height: '100',
          size: '326593',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100.mp4',
          mp4_size: '33121',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100.webp',
          webp_size: '95102'
        },
        downsized_still: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy_s.gif',
          width: '275',
          height: '228'
        },
        downsized: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.gif',
          width: '275',
          height: '228',
          size: '1277013'
        },
        downsized_large: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.gif',
          width: '275',
          height: '228',
          size: '1277013'
        },
        fixed_width_small_still: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100w_s.gif',
          width: '100',
          height: '83'
        },
        preview_webp: {
          url:
            'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy-preview.webp',
          width: '145',
          height: '120',
          size: '49368'
        },
        fixed_width_still: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200w_s.gif',
          width: '200',
          height: '166'
        },
        fixed_width_small: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100w.gif',
          width: '100',
          height: '83',
          size: '231634',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100w.mp4',
          mp4_size: '26614',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/100w.webp',
          webp_size: '74706'
        },
        downsized_small: {
          width: '274',
          height: '228',
          mp4:
            'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy-downsized-small.mp4',
          mp4_size: '131092'
        },
        fixed_width_downsampled: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200w_d.gif',
          width: '200',
          height: '166',
          size: '170389',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200w_d.webp',
          webp_size: '43140'
        },
        downsized_medium: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.gif',
          width: '275',
          height: '228',
          size: '1277013'
        },
        original: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.gif',
          width: '275',
          height: '228',
          size: '1277013',
          frames: '28',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.mp4',
          mp4_size: '254030',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.webp',
          webp_size: '351092'
        },
        fixed_height: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200.gif',
          width: '241',
          height: '200',
          size: '1088717',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200.mp4',
          mp4_size: '81947',
          webp: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/200.webp',
          webp_size: '252538'
        },
        looping: {
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy-loop.mp4',
          mp4_size: '1372569'
        },
        original_mp4: {
          width: '480',
          height: '396',
          mp4: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy.mp4',
          mp4_size: '254030'
        },
        preview_gif: {
          url: 'https://media3.giphy.com/media/ZXdPWPdzDTFlK/giphy-preview.gif',
          width: '81',
          height: '67',
          size: '48946'
        },
        '480w_still': {
          url: 'https://media4.giphy.com/media/ZXdPWPdzDTFlK/480w_s.jpg',
          width: '480',
          height: '398'
        }
      },
      title: 'john paul GIF',
      _score: 2500023.5,
      analytics: {
        onload: {
          url:
            'https://giphy_analytics.giphy.com/simple_analytics?response_id=5bf9b9906672516451c16e27&event_type=GIF_SEARCH&gif_id=ZXdPWPdzDTFlK&action_type=SEEN'
        },
        onclick: {
          url:
            'https://giphy_analytics.giphy.com/simple_analytics?response_id=5bf9b9906672516451c16e27&event_type=GIF_SEARCH&gif_id=ZXdPWPdzDTFlK&action_type=CLICK'
        },
        onsent: {
          url:
            'https://giphy_analytics.giphy.com/simple_analytics?response_id=5bf9b9906672516451c16e27&event_type=GIF_SEARCH&gif_id=ZXdPWPdzDTFlK&action_type=SENT'
        }
      }
    }
  ];

  window.fetch = jest.fn().mockImplementation(() => {
    Promise.resolve({
      json: () => Promise.resolve(mockGifs)
    });
  });

  beforeEach(() => {
    wrapper = shallow(<Gifs />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fetch gifs on CDM', () => {
    wrapper.instance().getGifs();
    expect(window.fetch).toHaveBeenCalledWith(
      'https://api.giphy.com/v1/gifs/search?q=paul&api_key=undefined&limit=200'
    );
  });

  it('should handle input change', () => {
    const mockEvent = { target: { value: 'cats' } };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state().input).toEqual('cats');
  });

  it('should render gifs', () => {});
});
