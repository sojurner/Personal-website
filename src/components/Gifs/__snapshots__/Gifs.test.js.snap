// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifs should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gifs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="jify"
        >
          PAUL
           LAND
        </h1>,
        <form
          className="gif-form"
          onSubmit={[Function]}
        >
          <input
            className="gif-input"
            onChange={[Function]}
            placeholder="Search Gifs"
            type="text"
            value=""
          />
          <i
            className="fab fa-searchengin"
            onClick={[Function]}
          />
        </form>,
        "",
        <div
          id="scrolling"
        >
          <InfiniteScroll
            className="infinite-scroll"
            dataLength={0}
            disableBrowserPullToRefresh={true}
            hasMore={true}
            loader={
              <h4>
                Loading...
              </h4>
            }
            next={[Function]}
            pullDownToRefreshContent={
              <h3>
                Pull down to refresh
              </h3>
            }
            pullDownToRefreshThreshold={100}
            releaseToRefreshContent={
              <h3>
                Release to refresh
              </h3>
            }
            scrollableTarget="scrolling"
          />
        </div>,
      ],
      "className": "gif-container",
      "id": "gifs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "PAUL",
            " LAND",
          ],
          "className": "jify",
        },
        "ref": null,
        "rendered": Array [
          "PAUL",
          " LAND",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="gif-input"
              onChange={[Function]}
              placeholder="Search Gifs"
              type="text"
              value=""
            />,
            <i
              className="fab fa-searchengin"
              onClick={[Function]}
            />,
          ],
          "className": "gif-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "gif-input",
              "onChange": [Function],
              "placeholder": "Search Gifs",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fab fa-searchengin",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "form",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <InfiniteScroll
            className="infinite-scroll"
            dataLength={0}
            disableBrowserPullToRefresh={true}
            hasMore={true}
            loader={
              <h4>
                Loading...
              </h4>
            }
            next={[Function]}
            pullDownToRefreshContent={
              <h3>
                Pull down to refresh
              </h3>
            }
            pullDownToRefreshThreshold={100}
            releaseToRefreshContent={
              <h3>
                Release to refresh
              </h3>
            }
            scrollableTarget="scrolling"
          />,
          "id": "scrolling",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "className": "infinite-scroll",
            "dataLength": 0,
            "disableBrowserPullToRefresh": true,
            "hasMore": true,
            "loader": <h4>
              Loading...
            </h4>,
            "next": [Function],
            "pullDownToRefreshContent": <h3>
              Pull down to refresh
            </h3>,
            "pullDownToRefreshThreshold": 100,
            "releaseToRefreshContent": <h3>
              Release to refresh
            </h3>,
            "scrollableTarget": "scrolling",
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="jify"
          >
            PAUL
             LAND
          </h1>,
          <form
            className="gif-form"
            onSubmit={[Function]}
          >
            <input
              className="gif-input"
              onChange={[Function]}
              placeholder="Search Gifs"
              type="text"
              value=""
            />
            <i
              className="fab fa-searchengin"
              onClick={[Function]}
            />
          </form>,
          "",
          <div
            id="scrolling"
          >
            <InfiniteScroll
              className="infinite-scroll"
              dataLength={0}
              disableBrowserPullToRefresh={true}
              hasMore={true}
              loader={
                <h4>
                  Loading...
                </h4>
              }
              next={[Function]}
              pullDownToRefreshContent={
                <h3>
                  Pull down to refresh
                </h3>
              }
              pullDownToRefreshThreshold={100}
              releaseToRefreshContent={
                <h3>
                  Release to refresh
                </h3>
              }
              scrollableTarget="scrolling"
            />
          </div>,
        ],
        "className": "gif-container",
        "id": "gifs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "PAUL",
              " LAND",
            ],
            "className": "jify",
          },
          "ref": null,
          "rendered": Array [
            "PAUL",
            " LAND",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="gif-input"
                onChange={[Function]}
                placeholder="Search Gifs"
                type="text"
                value=""
              />,
              <i
                className="fab fa-searchengin"
                onClick={[Function]}
              />,
            ],
            "className": "gif-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "gif-input",
                "onChange": [Function],
                "placeholder": "Search Gifs",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fab fa-searchengin",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "form",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <InfiniteScroll
              className="infinite-scroll"
              dataLength={0}
              disableBrowserPullToRefresh={true}
              hasMore={true}
              loader={
                <h4>
                  Loading...
                </h4>
              }
              next={[Function]}
              pullDownToRefreshContent={
                <h3>
                  Pull down to refresh
                </h3>
              }
              pullDownToRefreshThreshold={100}
              releaseToRefreshContent={
                <h3>
                  Release to refresh
                </h3>
              }
              scrollableTarget="scrolling"
            />,
            "id": "scrolling",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [],
              "className": "infinite-scroll",
              "dataLength": 0,
              "disableBrowserPullToRefresh": true,
              "hasMore": true,
              "loader": <h4>
                Loading...
              </h4>,
              "next": [Function],
              "pullDownToRefreshContent": <h3>
                Pull down to refresh
              </h3>,
              "pullDownToRefreshThreshold": 100,
              "releaseToRefreshContent": <h3>
                Release to refresh
              </h3>,
              "scrollableTarget": "scrolling",
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
